### PR TITLE
Fix #25860: Add Theme Column for Template Listing

### LINKED ISSUE #25860: Template Builder: Add a Link to Themes in listing
### User Story
As a dotCMS user, I want to be able to link to themes easily, so I can quickly find and access them.

### Acceptance Criteria
- A link to the theme folder in the template listing and template detail should be added.
- The link should be easy to find and navigate to.

### Proposed Objective
Core Feature

### Proposed Priority
Priority 3 - Average

### External Links... Slack Conversations, Support Tickets, Figma Designs, etc.:
Image

This link to the Site Browser to the corresponding theme folder open, we might need some backend work for this.

### Assumptions & Initiation Needs
- The link must be added in a way that does not interfere with the existing navigation of the template listing and builder.
- We need extra backend work to be able to link to the folder like we did in the containers portlet

### Quality Assurance Notes & Workarounds
N/A

### Sub-Tasks & Estimates:
- Create a link to the theme folder in the template listing and template detail: 2 hours

### PR DESCRIPTION
**Pull request by zJaaal**
### Summary
🎨🗂️🧪

This pull request adds a new feature to the dot-template-list component, which shows the theme name and link for each template in a new column. It also enables folder navigation from the template list. It updates the dot-template and dot-theme models, the component HTML, TS, and spec files, and the language properties file to support this feature.

> _We're sailing on the web with our templates and themes_
> _We're adding new features to our dotCMS team_
> _We'll pull and push and merge with a hearty yo-ho-ho_
> _And we'll test our `dot-template-list` before we go_

### Walkthrough
*  Add theme column to template list table and display template's theme name or link to theme folder ([link](https://github.com/dotCMS/core/pull/26129/files?diff=unified&w=0#diff-ce14bb21a1834f2bb62ae18a86c955b285337e3a34381ba1ce0f40439a35558fL58-R74), [link](https://github.com/dotCMS/core/pull/26129/files?diff=unified&w=0#diff-e16e5d71145ece8e7e3ac84c6efb9c108a31b1ccd8fee53dd7a309f71b295aa3R331-R334), [link](https://github.com/dotCMS/core/pull/26129/files?diff=unified&w=0#diff-edc803a51d566571d1691c20d1928c9a5b702990f6b595bc878d05da267e0ee8R263-R266), [link](https://github.com/dotCMS/core/pull/26129/files?diff=unified&w=0#diff-21db9723fa4c52389731b7575f1e19bd364aa992b7e549e22c08a33456994e94R5119))
*  Add themeInfo property to DotTemplate interface and DotTheme interface to store theme metadata ([link](https://github.com/dotCMS/core/pull/26129/files?diff=unified&w=0#diff-573769f19f94628ab80de15424cc69d7c7b5d5902e41b9461658eb09d7e6dd66R3), [link](https://github.com/dotCMS/core/pull/26129/files?diff=unified&w=0#diff-573769f19f94628ab80de15424cc69d7c7b5d5902e41b9461658eb09d7e6dd66R35), [link](https://github.com/dotCMS/core/pull/26129/files?diff=unified&w=0#diff-538f3ce8cb27bb085beaaa42c6f18582a404e710c006c4ac0c95c171e278f595R13-R19))
*  Add goToFolder method to dot-template-list component to navigate to theme folder when clicking on theme link ([link](https://github.com/dotCMS/core/pull/26129/files?diff=unified&w=0#diff-edc803a51d566571d1691c20d1928c9a5b702990f6b595bc878d05da267e0ee8R207-R221))
*  Add test cases for theme column rendering, theme link functionality, system theme handling, and empty theme case ([link](https://github.com/dotCMS/core/pull/26129/files?diff=unified&w=0#diff-e16e5d71145ece8e7e3ac84c6efb9c108a31b1ccd8fee53dd7a309f71b295aa3R532-R581))
*  Add theme and themeInfo properties to template mock objects for testing purposes ([link](https://github.com/dotCMS/core/pull/26129/files?diff=unified&w=0#diff-e16e5d71145ece8e7e3ac84c6efb9c108a31b1ccd8fee53dd7a309f71b295aa3L79-R100), [link](https://github.com/dotCMS/core/pull/26129/files?diff=unified&w=0#diff-e16e5d71145ece8e7e3ac84c6efb9c108a31b1ccd8fee53dd7a309f71b295aa3L97-R139), [link](https://github.com/dotCMS/core/pull/26129/files?diff=unified&w=0#diff-e16e5d71145ece8e7e3ac84c6efb9c108a31b1ccd8fee53dd7a309f71b295aa3L114-R177), [link](https://github.com/dotCMS/core/pull/26129/files?diff=unified&w=0#diff-e16e5d71145ece8e7e3ac84c6efb9c108a31b1ccd8fee53dd7a309f71b295aa3L130-R214), [link](https://github.com/dotCMS/core/pull/26129/files?diff=unified&w=0#diff-e16e5d71145ece8e7e3ac84c6efb9c108a31b1ccd8fee53dd7a309f71b295aa3L147-R270))
*  Remove unnecessary line breaks in component tags ([link](https://github.com/dotCMS/core/pull/26129/files?diff=unified&w=0#diff-ce14bb21a1834f2bb62ae18a86c955b285337e3a34381ba1ce0f40439a35558fL15-R15), [link](https://github.com/dotCMS/core/pull/26129/files?diff=unified&w=0#diff-ce14bb21a1834f2bb62ae18a86c955b285337e3a34381ba1ce0f40439a35558fL23-R22), [link](https://github.com/dotCMS/core/pull/26129/files?diff=unified&w=0#diff-ce14bb21a1834f2bb62ae18a86c955b285337e3a34381ba1ce0f40439a35558fL33-R31), [link](https://github.com/dotCMS/core/pull/26129/files?diff=unified&w=0#diff-ce14bb21a1834f2bb62ae18a86c955b285337e3a34381ba1ce0f40439a35558fL46-R43), [link](https://github.com/dotCMS/core/pull/26129/files?diff=unified&w=0#diff-ce14bb21a1834f2bb62ae18a86c955b285337e3a34381ba1ce0f40439a35558fL74-R83), [link](https://github.com/dotCMS/core/pull/26129/files?diff=unified&w=0#diff-ce14bb21a1834f2bb62ae18a86c955b285337e3a34381ba1ce0f40439a35558fL84-R92))

## Screenshots

https://github.com/dotCMS/core/assets/63567962/b01edbd8-4376-4b7f-a25f-217af887417d

### CODE REVIEW

**Review by oidacra — APPROVED** [review]

File: core-web/apps/dotcms-ui/src/app/portlets/dot-templates/dot-template-list/dot-template-list.component.spec.ts [hunk 1]
```diff
@@ -401,6 +529,57 @@ describe('DotTemplateListComponent', () => {
...
+        });
+
+        it('should trigger goToFolder whem clicking on a theme link', () => {
+            const mockGoToFolder = spyOn(comp, 'goToFolder');
+
+            const link = fixture.debugElement.query(By.css('[data-testid="theme-folder-link"]'));
+
+            link.nativeElement.click();
+
+            expect(mockGoToFolder).toHaveBeenCalled();
```
Comment (fmontes) [inline core-web/apps/dotcms-ui/src/app/portlets/dot-templates/dot-template-list/dot-template-list.component.spec.ts:558]:
```suggestion
            expect(mockGoToFolder).toHaveBeenCalled();
```
```suggestion
            expect(mockGoToFolder).toHaveBeenCalledWith(xx, xx, xxx);
```

File: core-web/apps/dotcms-ui/src/app/portlets/dot-templates/dot-template-list/dot-template-list.component.spec.ts [hunk 2]
```diff
@@ -401,6 +529,57 @@ describe('DotTemplateListComponent', () => {
...
+        });
+
+        it("should render 'System Theme' when the theme is SYSTEM_THEME", () => {
+            const mockGoToFolder = spyOn(comp, 'goToFolder');
+            const cells = fixture.debugElement.queryAll(By.css('[data-testid="theme-cell"]'));
+
+            cells[1].nativeElement.click();
+            expect(cells[1].nativeElement.textContent.trim()).toEqual('System Theme');
+
+            expect(mockGoToFolder).not.toHaveBeenCalled();
```
Comment (fmontes) [inline core-web/apps/dotcms-ui/src/app/portlets/dot-templates/dot-template-list/dot-template-list.component.spec.ts:568]: System theme have no folder, so should not have link

File: core-web/apps/dotcms-ui/src/app/portlets/dot-templates/dot-template-list/dot-template-list.component.spec.ts [hunk 3]
```diff
@@ -401,6 +529,57 @@ describe('DotTemplateListComponent', () => {
...
+        it('should render empty when the theme is undefined or null', () => {
+            const mockGoToFolder = spyOn(comp, 'goToFolder');
+            const cells = fixture.debugElement.queryAll(By.css('[data-testid="theme-cell"]'));
+
+            const lastCell = cells.pop();
+
+            lastCell.nativeElement.click();
+            expect(lastCell.nativeElement.textContent.trim()).toEqual('');
+
+            expect(mockGoToFolder).not.toHaveBeenCalled();
```
Comment (fmontes) [inline core-web/apps/dotcms-ui/src/app/portlets/dot-templates/dot-template-list/dot-template-list.component.spec.ts:580]: If you are only testing the render, why are you doing the click?

File: core-web/apps/dotcms-ui/src/app/portlets/dot-templates/dot-template-list/dot-template-list.component.ts [hunk 4]
```diff
@@ -203,6 +203,21 @@ export class DotTemplateListComponent implements OnInit, OnDestroy {
...
+     * Set the selected folder in the Site Browser portlet.
+     *
+     * @param {Event} event
+     * @param {string} path
+     * @memberof DotTemplateListComponent
+     */
+    goToFolder(event: Event, path: string) {
+        event.stopPropagation();
+
+        this.dotSiteBrowserService.setSelectedFolder(path).subscribe(() => {
```
Comment (fmontes) [inline core-web/apps/dotcms-ui/src/app/portlets/dot-templates/dot-template-list/dot-template-list.component.ts:216]:
```suggestion
        this.dotSiteBrowserService.setSelectedFolder(path).pipe(take(1)).subscribe(() => {
```

**Review by fmontes — CHANGES_REQUESTED** [review]

File: core-web/apps/dotcms-ui/src/app/portlets/dot-templates/dot-template-list/dot-template-list.component.ts (hunk 4 above)
Comment (zJaaal) [reply core-web/apps/dotcms-ui/src/app/portlets/dot-templates/dot-template-list/dot-template-list.component.ts:216]: That method takes one under the hood

**Review by fmontes — APPROVED** [review]
